src/librustc/mir/mod.rs

```diff
@@ -467,7 +467,9 @@ impl<T: Decodable> rustc_serialize::UseSpecializedDecodable for ClearCrossCrate<
 /// Grouped information about the source code origin of a MIR entity.
 /// Intended to be inspected by diagnostics and debuginfo.
 /// Most passes can work with it as a whole, within a single function.
-#[derive(Copy, Clone, Debug, PartialEq, RustcEncodable, RustcDecodable, HashStable)]
+// The unoffical Cranelift backend, at least as of #65828, needs `SourceInfo` to implement `Eq` and
+// `Hash`. Please ping @bjorn3 if removing them.
```

Comment:
I don't find this kind of comment particularly useful. Also, I would expect you'd use an `IndexVec<SourceScope, ...>` instead of working with `SourceInfo` directly.

Reply:
Comment was suggested at https://github.com/rust-lang/rust/pull/65828#issuecomment-546484520. I am getting the `SourceInfo` directly from `Statement` and `Terminator`, but I need to get an unique u32 to pass through `cranelift` as `SourceLoc`, while being able to get the original `SourceInfo` back after having compiled a function.

Reply:
Hmm, unless you are emitting variable debuginfo somehow, you'd only need the `Span`, not the whole `SourceInfo`.

Reply:
> Hmm, unless you are emitting variable debuginfo somehow

Not yet

> you'd only need the Span, not the whole SourceInfo.

Indeed only `Span` is used currently, but I will probably need the rest of the `SourceInfo`.